server/src/db/queries/pathway/modifyPathwaySteps.js

```diff
@@ -0,0 +1,119 @@
+const Query = require('../../core/Query')
+const Model = require('../../core/Model')
+
+const convertToProp = (step) => {
+    let prop = {
+        ...step,
+        stepDataId: Model.generateId('StepData')
+    }
+
+    if (!step.isPathway) {
+        prop.stepContentId = Model.generateId('StepContent')
+    } 
+
+    delete step.isPathway
+    return prop
+}
+
+const createPathwaySteps = ({id, createSteps}) => {
+    const props = {
+        pathwayProps: [],
+        contentProps: []
+    }
+
+    createSteps.forEach((step) => {
+        step.isPathway ? props.pathwayProps.push(convertToProp(step))
+                       : props.contentProps.push(convertToProp(step))
+    })
+
+    console.log("Creating new steps ran")
+    console.log(props)
+
+    return new Query({
+        statement: `
+        MATCH (p:Pathway {id: $id})
+            FOREACH (prop IN $contentProps | 
+                CREATE (sd:Step {id: prop.stepDataId, name: prop.name, difficulty: prop.difficulty, index: prop.index, isPathway: false})
+                CREATE (sc:Content {id: prop.stepContentId, content: prop.content})
+                MERGE (p)<-[:HAS_PARENT_PATHWAY]-(sd)-[:HAS_CONTENT]->(sc)
+            )
+        WITH p
+            FOREACH (prop IN $pathwayProps | 
+                CREATE (sd:Step {id: prop.stepDataId, name: prop.name, difficulty: prop.difficulty, index: prop.index, isPathway: true})
+                MERGE (p1:Pathway {id: prop.pathwayId})
+                MERGE (p)<-[:HAS_PARENT_PATHWAY]-(sd)-[:INCLUDES]->(p1)
+            )
+        RETURN p
+        `,
+        params: {
+            pathwayProps: props.pathwayProps,
+            contentProps: props.contentProps,
+            id,
+        }
+    })
+}
+
+const updatePathwaySteps = ({id, updateSteps}) => {
+    const steps = []
+    updateSteps.forEach((step) => {
+        steps.push(convertToProp(step));
+    })
+
+    console.log("Updating steps")
+    console.log(steps)
+
+    return new Query({
+        statement: `
+        MATCH (p:Pathway {id: $id})
+        WITH p
+        UNWIND $steps as upd
+        WITH p, upd
+        MATCH (p)<-[:HAS_PARENT_PATHWAY]-(s:Step {id: upd.id})
+        SET s.name = upd.name, s.difficulty = upd.difficulty, s.index = upd.index
```

Comment:
This query does not allow updating the step content, even though it's a part of `StepPayload`. Are you going to write a different endpoint for updating content?

Reply:
Yes I am assuming three layers,
1. Raw data - md file - step payload
2. Step (can be pathway) - meta data - time complete
3. Pathway (made of steps) - additional meta data - order of steps

So each layers should have different endpoints.

Reply:
I have a problem with this approach. I see no reason why these layers should be decoupled for the client too. For the client, I think that a step should be an atomic entity. The client does not need to know how we're splitting a step into metadata and content.

Reply:
Let me try to explain my position better. I feel that here there are two types of data here,
1. The actual instructions to learn something
2. The interlinking between them using the concept of steps and pathways

These should necessarily be decoupled in UX as well as the APIs. The reason being that a user will interact with them separately. For the actual instructions an author will probably make a PR or edit it live. But this will only change the instructions to learn something, we can call it `Resource`. For the linking part, an author will create a step pointing to a resource (or an existing pathway) and link it to a new pathway. Here the author will probably use a graph view that allows them to easily search and link.

Another reason I feel these two types of data should have separate interfaces is to ensure the atomicity of the `Resource`. That is a `Resource` should exist standalone. It should not be created to suit the needs of one particular pathway. With this we can ensure that authors follow a certain flow:
1. Create a new `Resource` for a topic
2. See if a pathway for a particular skill can be created using existing `Resource` or pathways